contracts/token/ERC721/extensions/draft-ERC721Royalty.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.4.0 (token/ERC721/extensions/draft-ERC721Royalty.sol)
+
+pragma solidity ^0.8.0;
+
+import "./draft-IERC721Royalty.sol";
+import "../../../utils/introspection/ERC165Storage.sol";
+import "hardhat/console.sol";
+
+/**
+ * @dev Implementation of the ERC721 Royalty extension allowing royalty information to be stored and retrieved, as defined in
+ * https://eips.ethereum.org/EIPS/eip-2981[EIP-2981].
+ *
+ * Adds the {_setTokenRoyalty} methods to set the token royalty information, and {_setRoyalty} method to set a global
+ * royalty information.
+ *
+ * _Available since v4.5._
+ */
+abstract contract ERC721Royalty is IERC721Royalty, ERC165Storage {
+    struct RoyaltyInfo {
+        address receiver;
+        uint256 royaltyAmount;
+    }
+
+    RoyaltyInfo private _royaltyInfo;
+    mapping(uint256 => RoyaltyInfo) private _tokenRoyalty;
+
+    /*
+     * @dev Sets tokens royalties
+     *
+     * Requirements:
+     * - `tokenId` must be already mined.
+     * - `recipient` cannot be the zero address.
+     * - `value` must indicate the percentage value using two decimals.
+     */
+    function _setTokenRoyalty(
```

Comment:
There is no way to delete token royalty if it needs to be reset back to global royalty. We might need a separate function to do that.